src/coreclr/tools/superpmi/mcs/verbmerge.cpp

```diff
@@ -281,12 +288,57 @@ int verbMerge::FilterDirectory(LPCWSTR                      dir,
         dirent *pEntry = readdir(pDir);
         while (pEntry != nullptr)
         {
-            if ((fnmatch(searchPatternUtf8.c_str(), pEntry->d_name, 0) == 0) && filter(pEntry))
+            struct stat sb;
+            int dirEntryType;
+
+#if HAVE_DIRENT_D_TYPE
+            dirEntryType = pEntry->d_type;
+#else
+            dirEntryType = DT_UNKNOWN;
+#endif
+	    // On some systems, dirent contains a d_type field, but note:
+	    // some file systems MAY leave d_type == DT_UNKNOWN,
+	    // expecting the consumer to stat the file. On systems
+	    // without a d_type simply always stat the file.
+
+	    if (dirEntryType == DT_UNKNOWN)
+	    {
+
+		if (fstatat(dirfd(pDir), pEntry->d_name, &sb, 0) == -1)
+		{
+		    continue;
+		}
+
+		if (S_ISDIR(sb.st_mode)) {
+		    dirEntryType = DT_DIR;
+		} else if (S_ISREG(sb.st_mode)) {
+		    dirEntryType = DT_REG;
+		} else if (S_ISLNK(sb.st_mode)) {
+		    dirEntryType = DT_LNK;
+		} else {
+		    dirEntryType = DT_UNKNOWN;
+		}
+	    }
+
+	    if (dirEntryType == DT_UNKNOWN)
```

Comment:
nit: this file uses four-space indentation, your delta has mixed tabs and spaces in this file.
```suggestion
            // On some systems, dirent contains a d_type field, but note:
            // some file systems MAY leave d_type == DT_UNKNOWN,
            // expecting the consumer to stat the file. On systems
            // without a d_type simply always stat the file.

            if (dirEntryType == DT_UNKNOWN)
            {
                if (fstatat(dirfd(pDir), pEntry->d_name, &sb, 0) == -1)
                    continue;

                if (S_ISDIR(sb.st_mode))
                    dirEntryType = DT_DIR;
                else if (S_ISREG(sb.st_mode))
                    dirEntryType = DT_REG;
                else if (S_ISLNK(sb.st_mode))
                    dirEntryType = DT_LNK;
                else
                    dirEntryType = DT_UNKNOWN;
            }

            if (dirEntryType == DT_UNKNOWN)
```

Reply:
I think I have these the way you suggested now.